src/test/functional/functional-tests.js

```diff
@@ -1754,4 +1754,101 @@ describe('functional tests', function() {
     })
 
   })
+
+  describe('Object retention API Tests', ()=>{
+    //Isolate the bucket/object for easy debugging and tracking.
+    //Gateway mode does not support this header.
+
+    describe('Object retention get/set API Test', function () {
+      const  objRetentionBucket = "minio-js-test-retention-" + uuid.v4()
+      const retentionObjName = "RetentionObject"
+      let isFeatureSupported = false
+      let versionId = null
+
+      step(`Check if bucket with object lock can be created:_bucketName:${objRetentionBucket}`, done => {
+        client.makeBucket(objRetentionBucket, { ObjectLocking: true }, (err) => {
+          if (err && err.code === 'NotImplemented') return done()
+          isFeatureSupported= true
+          if (err) return done(err)
+          done()
+        })
+      })
+
+      step(`putObject(bucketName, objectName, stream)_bucketName:${objRetentionBucket}, objectName:${retentionObjName}, stream:100Kib_`, done => {
+        //Put two versions of the same object.
+        if(isFeatureSupported) {
+          client.putObject(objRetentionBucket, retentionObjName,  readableStream(_1byte), _1byte.length, {})
+            .then(() => done())
+            .catch(done)
+        }else{
+          done()
+        }
+      })
+
+      step(`statObject(bucketName, objectName, statOpts)_bucketName:${objRetentionBucket}, objectName:${retentionObjName}`, done => {
+        if(isFeatureSupported) {
+          client.statObject(objRetentionBucket, retentionObjName, {}, (e, res) => {
+            versionId = res.versionId
+            done()
+          })
+        }else{
+          done()
+        }
+
+      })
+
+      step(`putObjectRetention(bucketName, objectName, putOpts)_bucketName:${objRetentionBucket}, objectName:${retentionObjName}`, done => {
+        //Put two versions of the same object.
+        if(isFeatureSupported) {
+          const expirationDate = new Date('2021-08-02')
```

Comment:
Can you get the current time and add a say or so, instead of hard coding the date here?

Reply:
Thank you :+1:  . I have updated the date to be dynamic.